include/Utils/DIPUtils.h

```diff
@@ -555,6 +661,179 @@ void BilinearInterpolationResizing(
       });
 }
 
+// Inserts a constant op with value -1 into a location `loc` based on type
+// `type`. Supported types are : f32, f64, integer types
+Value insertMinusOneConstantOp(MLIRContext *ctx, OpBuilder &builder,
+                               Location loc, Type elemTy) {
+  Value op = {};
+  auto bitWidth = elemTy.getIntOrFloatBitWidth();
+  if (elemTy.isF32() || elemTy.isF64()) {
+    FloatType type =
+        elemTy.isF32() ? FloatType::getF32(ctx) : FloatType::getF64(ctx);
+    op = elemTy.isF32()
+             ? builder.create<ConstantFloatOp>(loc, (APFloat)(float)-1, type)
+             : builder.create<ConstantFloatOp>(loc, (APFloat)(double)-1, type);
+  } else if (elemTy.isInteger(bitWidth)) {
+    IntegerType type = IntegerType::get(ctx, bitWidth);
+    op = builder.create<ConstantIntOp>(loc, -1, type);
+  }
+
+  return op;
+}
+
+// Util function for morphological transformations; used to initially fill the
+// output memref.
+Value createMaskVecOp(OpBuilder &builder, Location loc, VectorType type,
+                      Value outputVec, Value paddingVec) {
+  Value maskVec = {};
+  auto elemTy = type.getElementType();
+  auto bitWidth = elemTy.getIntOrFloatBitWidth();
+  if (elemTy.isF32() || elemTy.isF64()) {
+    maskVec =
+        builder.create<CmpFOp>(loc, CmpFPredicate::OGT, outputVec, paddingVec);
+  } else if (elemTy.isInteger(bitWidth)) {
+    maskVec =
+        builder.create<CmpIOp>(loc, CmpIPredicate::sgt, outputVec, paddingVec);
+  }
+  return maskVec;
+}
+
+// Util function for morphological transformations ; compares two vectors and
+// returns a mask
+Value createCompVecMorph(OpBuilder &builder, Location loc, VectorType type,
+                         Value inputVec, Value outputVec, DIP_OP op) {
+  Value compVec = {};
+  auto elemTy = type.getElementType();
+  auto bitWidth = elemTy.getIntOrFloatBitWidth();
+  if (elemTy.isF32() || elemTy.isF64()) {
+    if (op == DIP_OP::EROSION_2D) {
+      compVec =
+          builder.create<CmpFOp>(loc, CmpFPredicate::OGE, inputVec, outputVec);
+    } else if (op == DIP_OP::DILATION_2D) {
+      compVec =
+          builder.create<CmpFOp>(loc, CmpFPredicate::OLE, inputVec, outputVec);
+    }
+  } else if (elemTy.isInteger(bitWidth)) {
+    if (op == DIP_OP::EROSION_2D) {
+      compVec =
+          builder.create<CmpIOp>(loc, CmpIPredicate::sge, inputVec, outputVec);
+    } else if (op == DIP_OP::DILATION_2D) {
+      compVec =
+          builder.create<CmpIOp>(loc, CmpIPredicate::sle, inputVec, outputVec);
+    }
+  }
+  return compVec;
+}
+
+// Util function which help determine the nullity of a kernel element.
+Value zeroCond(OpBuilder &builder, Location loc, Type elemType,
```

Comment:
I have added this mechanism on a higher level (before any calculation for extrapolation happens) inside `traverseImagewBoundaryExtrapolation()`. Please change this in your PR appropriately.